### PR TITLE
Perf 2393 adding options to control loading behavior

### PR DESCRIPTION
**Pull request by ruchen**
Hi Greg, 

Could you please take a look at these later changes to add some options to sharded time series loading? 

So far these are 
        collectionSharded    bool                                                                                                                                                                                     
        numInitChunks        uint                                                                                                                                                                                     
        shardKeySpec         string                                                                                                                                                                                   
        balancerOn       bool                                                                                                                                                                                     

Thank you!

### CODE REVIEW

File: cmd/tsbs_load_mongo/main.go [hunk 1]
```diff
@@ -73,15 +79,14 @@ func init() {
 	retryableWrites = viper.GetBool("retryable-writes")
 	orderedInserts = viper.GetBool("ordered-inserts")
 	randomFieldOrder = viper.GetBool("random-field-order")
-	timeseriesCollectionSharded = viper.GetBool("timeseries-collection-sharded")
+	collectionSharded = viper.GetBool("collection-sharded")
+	numInitChunks = viper.GetUint("number-initial-chunks")
+	shardKeySpec = viper.GetString("shard-key-spec")
+	loadBalancerOn = viper.GetBool("load-balancer-on")
```
Comment (banarun) [inline cmd/tsbs_load_mongo/main.go:85]: nit: Suggest replacing `load balancer` with `balancer` to be consistent with the terminology that we use everywhere else in the code and docs.

File: cmd/tsbs_load_mongo/creator.go [hunk 2]
```diff
@@ -61,36 +61,57 @@ func (d *dbCreator) CreateDB(dbName string) error {
...
+		   	cmd2 = append(cmd2, bson.E{"numInitialChunks", numInitChunks})
+		}
+	   	res2 := d.client.Database("admin").RunCommand(context.Background(), cmd2)
+	
 		if res2.Err() != nil {
-			return fmt.Errorf("shard collection err: %v", res2.Err().Error())
+		        return fmt.Errorf("shard collection err: %v", res2.Err().Error())
+	        }
+
+		cmd3 := make(bson.D, 0, 4)
```
Comment (banarun) [inline cmd/tsbs_load_mongo/creator.go:102]: Can you rename `cmd1`, `cmd2` and `cmd3` with `enableShardingCmd`, `shardCollCmd` and `balancerCmd` to make the code a bit more readable?

**Review by banarun — APPROVED** [review]
LGTM! Thanks for adding more options to this, Rushan.

File: cmd/tsbs_load_mongo/creator.go (hunk 2 above)
Comment (ruchen) [reply cmd/tsbs_load_mongo/creator.go]: done.

File: scripts/load_mongo.sh [hunk 3]
```diff
@@ -14,6 +14,12 @@ MONGO_URL=${MONGO_URL:-"mongodb://username:password@localhost:27017/admin"}
 # Load parameters - personal
 PROGRESS_INTERVAL=${PROGRESS_INTERVAL:-10s}
 
+#default to timeseries_collection_sharded to false
+COLLECTION_SHARDED=${COLLECTION_SHARDED:-false}
+NUMBER_INITIAL_CHUNKS=${NUMBER_INITIAL_CHUNKS:-0}
+SHARD_KEY_SPEC=${SHARD_KEY_SPEC:-"\"time\":1"}
```
[the remark this answers is not in the document]
Comment (ruchen) [reply scripts/load_mongo.sh]: done

File: cmd/tsbs_load_mongo/main.go (hunk 1 above)
Comment (ruchen) [reply cmd/tsbs_load_mongo/main.go]: done

File: cmd/tsbs_load_mongo/creator.go [hunk 4]
```diff
@@ -50,47 +50,69 @@ func (d *dbCreator) RemoveOldDB(dbName string) error {
 }
 
 func (d *dbCreator) CreateDB(dbName string) error {
-	cmd := make(bson.D, 0, 4)
-	cmd = append(cmd, bson.E{"create", collectionName})
+	createTSCmd := make(bson.D, 0, 4)
```
Comment (gregorynoma) [inline cmd/tsbs_load_mongo/creator.go:53]: Why rename this to `createTSCmd`? Seems like the new name would be inaccurate if `timeseriesCollection` is false. Maybe just `createCmd`?

File: cmd/tsbs_load_mongo/main.go [hunk 5]
```diff
@@ -52,8 +55,11 @@ func init() {
 	pflag.Bool("retryable-writes", true, "Whether to use retryable writes")
 	pflag.Bool("ordered-inserts", true, "Whether to use ordered inserts")
 	pflag.Bool("random-field-order", true, "Whether to use random field order")
-	pflag.Bool("timeseries-collection-sharded", false, "Whether to shard a time-series collection")
-
+	pflag.Bool("collection-sharded", false, "Whether to shard the collection")
+	pflag.Uint("number-initial-chunks", 0, "number of initial chunks to create and distribute of an empty collection; if 0 then do not specifiy it")
```
Comment (gregorynoma) [inline cmd/tsbs_load_mongo/main.go:59]: I'm not quite sure what "if 0 then do not specifiy it" is trying to say?

File: cmd/tsbs_load_mongo/creator.go (hunk 4 above)
Comment (ruchen) [reply cmd/tsbs_load_mongo/creator.go:53]: sure will rename it createCollCmd to be consistent with the shardCollCmd later

**Review by gregorynoma — APPROVED** [review]

**Pull request merged**